src/coreclr/vm/dacenumerablehash.h

```diff
@@ -206,11 +210,25 @@ class DacEnumerableHashTable
         return m_pBuckets;
     }
 
+    // our bucket table uses two extra slots - slot [0] contains the length of the table,
+    //                                         slot [1] will contain the next version of the table if it resizes
+    static const int SLOT_LENGTH = 0;
+    static const int SLOT_NEXT = 1;
+
+    static DWORD GetLength(DPTR(PTR_VolatileEntry) buckets)
+    {
+        return (DWORD)dac_cast<TADDR>(buckets[SLOT_LENGTH]);
+    }
+
+    static DPTR(PTR_VolatileEntry) GetNext(DPTR(PTR_VolatileEntry) buckets)
+    {
+        return (DPTR(PTR_VolatileEntry))dac_cast<TADDR>(buckets[SLOT_NEXT]);
```

Comment:
@hoyosjs - is this correct or from TADDR it must be another dac_cast ?

Reply:
Like:
```c
    static DPTR(PTR_VolatileEntry) GetNext(DPTR(PTR_VolatileEntry) buckets)
    {
        return dac_cast<DPTR(PTR_VolatileEntry)>(dac_cast<TADDR>(buckets[SLOT_NEXT]));
    }
```

Reply:
If I understand this right `buckets[SLOT_NEXT]` is pointing to a new hash table that's an array of linked lists? Then it's 

`return dac_cast<DPTR(PTR_VolatileEntry)>(dac_cast<TADDR>(buckets[SLOT_NEXT]));`

which can be simplified to

`return dac_cast<DPTR(PTR_VolatileEntry)>(buckets[SLOT_NEXT]);`

Reply:
Yes. The `buckets[SLOT_NEXT]` will be pointing to the next larger table, if resize happened while you are doing a lookup. It will be rare. It is possible though, even though extremely rare to see more than one new version. 
Every new one is 2x larger than previous, so there can't be too many.